pandas/tests/groupby/test_groupby.py

```diff
@@ -2134,3 +2134,12 @@ def test_groupby_column_index_name_lost_fill_funcs(func):
     result = getattr(df_grouped, func)().columns
     expected = pd.Index(["a", "b"], name="idx")
     tm.assert_index_equal(result, expected)
+
+
+def test_groupby_unit64_float_conversion():
+    #  GH: 30859 groupby converts unit64 to floats sometimes
+    test_value = 16148277970000000000
+    df = pd.DataFrame({"first": [1], "second": [1], "value": [test_value]})
+    df_grouped = df.groupby(["first", "second"])["value"].max()
+    assert df_grouped.values == test_value
```

Comment:
can you call df_grouped -> result
and then construct expected and use
tm.assert_frame_equal(result, expected)

also move the test somewhere around this test `test_groupby_dtype_inference_empty`